apps/wallet/src/ui/app/pages/home/index.tsx

```diff
@@ -24,11 +24,23 @@ const HomePage = ({ disableNavigation, limitToPopUpSize = true }: Props) => {
     const initChecking = useInitializedGuard(true);
     const lockedChecking = useLockedGuard(false);
     const guardChecking = initChecking || lockedChecking;
+    const [visibility, setVisibility] = useState(
+        document?.visibilityState || null
+    );
     const dispatch = useAppDispatch();
     useEffect(() => {
-        const sub = of(guardChecking)
+        const callback = () => {
```

Comment:
There's actually an edge case here where the visibility state changes between the component being called, and the effect being setup. You can avoid this issue by just immediately calling `setVisibility(document.visibilityState);` in the effect (which will be a no-op if the state hasn't changed anyway).